code/sonarr.py

```diff
@@ -131,47 +174,69 @@ def getSonarrTagId(tag_name):
     # if tag_id was not found, create it
     if tag_id is None:
         response = requests.post(
-            SONARRURL + 'v3/tag?apikey=' + SONARRAPIKEY, data=str(params).encode('utf-8'))
+            sonarr['URL'] + '/tag?' + sonarr['APIKEY'], data=str(params).encode('utf-8'))
         if response.status_code == 201:
             tag_id = response.json()['id']
     return tag_id
 
 
-def sendToSonarr(newShows, mapping, sonarrList):
+def indexSonarrList(sonarr, newShows, mapping, sonarrList):
     listToAdd = []
     for show in newShows:
         if LOGGING:
-            pr("Looking for ID for " + show['title'])
+                pr("Checking for existing mapping for " + show['title'])
         if show['anilistId'] in [i['anilistId'] for i in mapping]:
+            # Declare result in advance
+            result = False
             map = mapping[[i['anilistId']
                            for i in mapping].index(show['anilistId'])]
             pr(show['title'] + " is mapped to " +
                str(map['title']) + " season " + str(map['season']))
             # First check if show is in sonarrList (and therefore already in sonarr)
             if map['tmdb_or_tvdb_Id'] in [i['tvdbId'] for i in sonarrList]:
-                # mapped show was already in sonarr
-                result = sonarrList[[i['tvdbId']
-                                     for i in sonarrList].index(map['tmdb_or_tvdb_Id'])]
-                result['anilistId'] = map['anilistId']
-                result['season'] = map['season']
+                if RESPECTFUL_ADDING:
+                    if LOGGING:
+                        pr("Only looking respectfully at existing entry for " + show['title'])
+                else:
+                    # mapped show was already in sonarr
+                    result = sonarrList[[i['tvdbId']
+                                        for i in sonarrList].index(map['tmdb_or_tvdb_Id'])]
+                    result['anilistId'] = map['anilistId']
+                    result['season'] = map['season']
             else:
                 # Searching for mapped show by tvdbId
-                result = search("tvdb:" + str(map['tmdb_or_tvdb_Id']))
+                result = search(sonarr, "tvdb:" + str(map['tmdb_or_tvdb_Id']))
                 result['season'] = map['season']
                 result['anilistId'] = show['anilistId']
-            listToAdd.append(result)
+                if AUTO_FILL_MAPPING:
+                    addMapping(result)
+            if result:
+                listToAdd.append(result)
         else:
+            if LOGGING:
+                pr("Asking Sonarr for ID for " + show['title'])
             print("Searching for " + show['title'] + ' by title and year')
```

Comment:
should be pr instead of print. Since we're saying this, we can delete the IF Logging above.

Reply:
![image](https://github.com/noggl/AniPlanrr/assets/1676068/f832a232-2043-4577-85eb-6afa4a1c73fe)
Agreed.
I think I moved this output from somewhere else and didn't realize that made it redundant. I will add mention that it's searching Sonarr to it, though.